VcpkgInstallManifestDependencies task does not work well for projects building in parallel

**Describe the bug**
The <Delete> and <Touch> tasks interfere with each other when InstallManifestDependencies runs as part of parallel build.

**Environment**
- OS: windows
- Compiler: <irrelevant>

**To Reproduce**
Steps to reproduce the behavior:
1. run msbuild using scripts\buildsystems\msbuild\vcpkg.props/targets where multiple projects are built in parallel

**Expected behavior**
Build should succeed

**Failure logs** 
error message like:
error MSB3061: Unable to delete file "path\to\buildstamp". The process cannot access the file 'path\to\buildstamp' because it is being used by another process.

--- a/scripts/buildsystems/msbuild/vcpkg.targets
+++ b/scripts/buildsystems/msbuild/vcpkg.targets
@@ -150,7 +150,6 @@
     </PropertyGroup>
     <Message Text="Installing vcpkg dependencies to $(_ZVcpkgInstalledDir)" Importance="High" />
     <MakeDir Directories="$(_ZVcpkgInstalledDir)" />
-    <Delete Files="$(_ZVcpkgMSBuildStampFile)" />
     <Message Text="%22$(_ZVcpkgExecutable)%22 install $(_ZVcpkgHostTripletParameter) --x-wait-for-lock --triplet %22$(VcpkgTriplet)%22 --vcpkg-root %22$(_ZVcpkgRoot)\%22 %22--x-manifest-root=$(_ZVcpkgManifestRoot)\%22 %22--x-install-root=$(_ZVcpkgInstalledDir)\%22 $(VcpkgAdditionalInstallOptions)"
           Importance="High" />
     <Exec Command="%22$(_ZVcpkgExecutable)%22 install $(_ZVcpkgHostTripletParameter) --x-wait-for-lock --triplet %22$(VcpkgTriplet)%22 --vcpkg-root %22$(_ZVcpkgRoot)\%22 %22--x-manifest-root=$(_ZVcpkgManifestRoot)\%22 %22--x-install-root=$(_ZVcpkgInstalledDir)\%22 $(VcpkgAdditionalInstallOptions)"

